build.sbt

```diff
@@ -68,14 +68,7 @@ val commonSettings = Seq(
   Compile / doc / scalacOptions += "-no-link-warnings",
   sourcesInBase := false,
   // file headers
-  headerLicense := Some(HeaderLicense.ALv2("2009-2019", "Mathias Doenitz")),
-  // reformat main and test sources on compile
-  scalafmtOnCompile := true
-)
-
-lazy val crossSettings = Seq(
```

Comment:
Just a note regarding these settings because its a bit confusing, but scalafmt actually doesn't fundamentally work with sbt keys/settings. What I mean by this is that there are many ways to invoke scalafmt (cli, intellij, sbt, etc etc) and hence the source of truth for which sources scalafmt works on is the `.scalafmt.conf` file.

Also note that newer versions of scalafmt have the `project.layout` config i.e. https://github.com/apache/incubator-pekko/blob/main/.scalafmt.conf#L77 which will automatically handle the correct location of sources and dialects in a typical sbt project so I don't think this part is even necessary.

Reply:
Ok, thank you for this info!
I never really looked into the scalafmt mechanics in any more detail, so these improvements are much appreciated!